app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -67,8 +67,12 @@ def import_all(uuids = nil, options = {})
         end
       end
 
-      def import_for_repository(repository, force = false)
+      def import_for_repository(repository, force = false, full_index = false)
         ids = content_unit_class.ids_for_repository(repository.pulp_id)
+        # Rpms cannot change in Pulp so we do not index them if they are already present
+        # in our database. Errata and Package Groups can change in Pulp, so we index
+        # all of them in the repository on each sync.
+        ids -= repository.rpms.map(&:uuid) if self == Katello::Rpm && !full_index
         self.import_all(ids, :index_repository_association => false) if repository.content_view.default? || force
         self.sync_repository_associations(repository, ids) if self.manage_repository_association
```

Comment:
The sync_repository_associations needs to occur on the full list, not the trimmed list, otherwise it will remove everything but what is new.